code/projects/app/src/app/shared/image-comparison/image-comparison.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from "@angular/core";
+import { BehaviorSubject, combineLatest, filter, Observable } from "rxjs";
+
+@Injectable({
+  providedIn: "root",
+})
+export class ImageComparisonService {
+  protected left = new BehaviorSubject<ImageData | null>(null);
+  protected right = new BehaviorSubject<ImageData | null>(null);
```

Comment:
Rest of the code uses `undefined` instead of `null` here, but I guess it will be changed regardless when doing signals.